src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1268,80 +1290,121 @@ static bool IsRedirectToGet(HttpStatusCode code)
+                    // are treated as a standard -OutFile request. This also disables appending local file.
+                    Resume = new SwitchParameter(false);
 
-                WriteVerbose(WebCmdletStrings.WebMethodResumeFailedVerboseMsg);
+                    using (HttpRequestMessage requestWithoutRange = GetRequest(currentUri))
+                    {
+                        FillRequestStream(requestWithoutRange);
+                        long requestContentLength = 0;
+                        if (requestWithoutRange.Content != null)
+                        {
+                            requestContentLength = requestWithoutRange.Content.Headers.ContentLength.Value;
+                        }
 
-                // Disable the Resume switch so the subsequent calls to GetResponse() and FillRequestStream()
-                // are treated as a standard -OutFile request. This also disables appending local file.
-                Resume = new SwitchParameter(false);
+                        string reqVerboseMsg = String.Format(
+                            CultureInfo.CurrentCulture,
+                            WebCmdletStrings.WebMethodInvocationVerboseMsg,
+                            requestWithoutRange.Method,
+                            requestWithoutRange.RequestUri,
+                            requestContentLength);
+                        WriteVerbose(reqVerboseMsg);
 
-                using (HttpRequestMessage requestWithoutRange = GetRequest(currentUri))
+                        return GetResponse(client, requestWithoutRange, keepAuthorization);
+                    }
+                }
+
+                _resumeSuccess = response.StatusCode == HttpStatusCode.PartialContent;
+
+                if (ShouldRetry(response.StatusCode))
                 {
-                    FillRequestStream(requestWithoutRange);
-                    long requestContentLength = 0;
-                    if (requestWithoutRange.Content != null)
-                        requestContentLength = requestWithoutRange.Content.Headers.ContentLength.Value;
-
-                    string reqVerboseMsg = String.Format(CultureInfo.CurrentCulture,
-                        WebCmdletStrings.WebMethodInvocationVerboseMsg,
-                        requestWithoutRange.Method,
-                        requestWithoutRange.RequestUri,
-                        requestContentLength);
-                    WriteVerbose(reqVerboseMsg);
-
-                    return GetResponse(client, requestWithoutRange, keepAuthorization);
+                    string retryMessage = string.Format(
+                        CultureInfo.CurrentCulture,
+                        WebCmdletStrings.RetryVerboseMsg,
+                        RetryIntervalSec,
+                        response.StatusCode);
+
+                    WriteVerbose(retryMessage);
+                    Task.Delay(WebSession.RetryIntervalInSeconds * 1000).GetAwaiter().GetResult();
```

Comment:
is there some better way to do this that will be responsive to ctrl+alt+del? Currently, if you use the default 60 seconds, got an error endpoint without verbose it looks like powershell is hung because ctrl+alt+delete doesn't end the command.

Reply:
Added a cancellation token for the Delay. Ctrl+C will cancel the delay and the webrequest.